extensions/lifecycle/tasks/LifecycleTask.js

```diff
@@ -379,7 +414,7 @@ class LifecycleTask extends BackbeatTask {
      * @param {function} done - callback(error, data)
      * @return {undefined}
      */
-    _compareObject(bucketData, obj, rules, log, done) {
+    _compareAnObject(bucketData, obj, rules, log, done) {
```

Comment:
Since `_compareObject` is already singular it can be understood as applying to a single object, I don't think it adds clarity to rename to `_compareAnObject`